projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php

```diff
@@ -972,8 +972,8 @@ public function is_in_disallowed_list( $in_disallowed_list, $form = array() ) {
 	public function prepare_for_akismet( $form ) {
 		$form['comment_type'] = 'contact_form';
 		$form['user_ip']      = isset( $_SERVER['REMOTE_ADDR'] ) ? sanitize_text_field( wp_unslash( $_SERVER['REMOTE_ADDR'] ) ) : '';
-		$form['user_agent']   = isset( $_SERVER['HTTP_USER_AGENT'] ) ? sanitize_text_field( wp_unslash( $_SERVER['HTTP_USER_AGENT'] ) ) : '';
-		$form['referrer']     = isset( $_SERVER['HTTP_REFERER'] ) ? sanitize_text_field( wp_unslash( $_SERVER['HTTP_REFERER'] ) ) : '';
+		$form['user_agent']   = isset( $_SERVER['HTTP_USER_AGENT'] ) ? esc_url_raw( wp_unslash( $_SERVER['HTTP_USER_AGENT'] ) ) : '';
```

Comment:
This one isn't a URL. It's just arbitrary text, which we should probably pass through more or less unchanged.
```suggestion
		$form['user_agent']   = isset( $_SERVER['HTTP_USER_AGENT'] ) ? filter_var( wp_unslash( $_SERVER['HTTP_USER_AGENT'] ) ) : '';
```